contracts/button/lib.rs

```diff
@@ -527,7 +537,23 @@ pub mod button_game {
         }
 
         fn mint_reward(&self, to: AccountId, amount: Balance) -> ButtonResult<()> {
-            PSP22MintableRef::mint(&self.data.get().unwrap().reward_token, to, amount)?;
+            let data = self.data.get().unwrap();
+
+            // scale the amount to always pay out full token units
+            let scaled_amount = amount.saturating_mul(1_000_000_000_000);
+
+            PSP22MintableRef::mint(&data.reward_token, to, scaled_amount)?;
+
+            Self::emit_event(
```

Comment:
Nit: I think mixing `self.` and `Self::` within one method is kinda ugly (and probably confusing to people who try to learn by reading such code). I'd propose to use `Self::` only where you really have to (constructor)

Reply:
IMHO it's not possible to use self for `emit_event` either - it's a static method (i.e. first argument is not `&self` )